batch/batch/driver/instance_collection/pool.py

```diff
@@ -493,4 +623,11 @@ async def schedule_with_error_handling(app, record, instance):
         if n_scheduled > 0:
             log.info(f'schedule: attempted to schedule {n_scheduled} jobs in {end - start}ms for {self.pool}')
 
+        if n_scheduled != n_attempted_to_schedule:
+            SCHEDULING_LOOP_RUNS_UNUSED_FREE_CORES.labels(pool_name=self.pool.name).inc()
+
+        pool_labels = PoolLabels(self.pool.name)
+        set_gauge_value(TOTAL_CORES_SCHEDULED, {pool_labels: n_cores_mcpu_scheduled / 1000})
+        set_gauge_value(TOTAL_CORES_ATTEMPTED_TO_SCHEDULE, {pool_labels: n_cores_mcpu_attempted_to_schedule / 1000})
```

Comment:
I think I find this function more confusing than it is worth. Can we just do:

```python
TOTAL_CORES_SCHEDULED.labels(pool_name=self.pool.name).set(n_cores_mcpu_scheduled / 1000)
```

I find this reads much more clearly. There's also no reason to clear these gauges before setting them